core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -315,14 +315,24 @@ public Data getData() {
         return data;
     }
 
+    /**
+     * Whether {@link #getData} might be blocking.
+     */
+    boolean hasUnparsedData() {
```

Comment:
It's not public but an explicit `@Restricted` might be a useful marker anyway. This looks like something we'd rather not expose unless there's a use case?

Reply:
I would just find it more confusing to place `@Restricted` on a non-`public` member. Ideally it would be a compile-time error.

Reply:
```suggestion
    // Internal use only
    boolean hasUnparsedData() {
```

Just want to prevent unintentionally opening this up in the interest of "cleanup" or similar.